algo1/api/generate.py

```diff
@@ -47,20 +50,36 @@ def generateSchedule(input: ScheduleConstraints):
 
   #Run algorithm on teacher preference matrix
   try:
-    output = random_search(matrix, avails)
+    output = run_random_search(courses, Times, avails, matrix)
+
   except Exception as e:
     logger.error(f"Failed generating Schedule: {e}")
     return None
 
   #Convert algorithm output to Schedule object
   try:
-    schedule = matrixToSchedule(output, profs, courses, courseMatcher, profMatcher, term)
+    schedule = tensorToSchedule(output, profs, courses, courseMatcher, profMatcher, term)
+    #schedule = matrixToSchedule(output, profs, courses, courseMatcher, profMatcher, term)
   except Exception as e:
     logger.error(f"Failed parsing generated schedule: {e}")
     return None
 
   return schedule
 
+def run_random_search(courses, times, avails, preferences):
+    card_c, card_ti, card_te = len(courses), len(times.items()), len(avails)
+    dims = {"courses":card_c, "times":card_ti, "teachers":card_te}
+    ro = RandOpt(dims, preferences, avails)
+
+    max_runtime = 600
+    start_time = time.time()
+    while (time.time() - start_time) < max_runtime:
+      ro = RandOpt(dims, preferences, avails)
+      ro.solve()
+      if ro.is_valid_schedule():
+        break
+
+    output = ro.sparse()
```

Comment:
I think we need to return the output here, I was testing with the api and it was failing to generate a schedule but it worked when I returned the output from run_random_search

Reply:
Good catch, just fixed it. Do you have access to the test data that was used yesterday?

Reply:
yeah, lemme test with that real quick

Reply:
It works on the test data file from company 3